fml/native_library.cc

```diff
@@ -0,0 +1,62 @@
+// Copyright 2017 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/fml/native_library.h"
+
+#include <dlfcn.h>
+#include <fcntl.h>
+
+namespace fml {
+
+NativeLibrary::NativeLibrary(const char* path) {
+  ::dlerror();
```

Comment:
I'm assuming we're not worried about thread safety with the error reset since our use of this is pretty limited and on one thread?

Reply:
`dlerror()` returns a per thread error status and the man pages actually advise that any call to `dl...` be preceded by a clearing of a potential error from other components. As written, this code is thread-safe.